### PR TITLE
feat(website): build the todomvc demo into the website build directory

### PR DESCRIPTION
**Pull request by KingHenne**
[Deploy-Preview](https://5c756b2d710dba000997bc5e--feature-hub.netlify.com/todomvc/)

### CODE REVIEW

File: packages/demos/src/todomvc/webpack-config.js [hunk 1]
```diff
@@ -1,46 +1,57 @@
-import CopyPlugin from 'copy-webpack-plugin';
-import {join} from 'path';
+// @ts-check
+const CopyPlugin = require('copy-webpack-plugin');
+const path = require('path');
```
Comment (clebert) [inline packages/demos/src/todomvc/webpack-config.js]: Why did you remove the destructuring everywhere?

File: packages/demos/src/webpack-base-config.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+// @ts-check
+const {join} = require('path');
+const TsConfigPathsPlugin = require('tsconfig-paths-webpack-plugin').default;
+const webpack = require('webpack');
 
-export const webpackBaseConfig: Configuration = {
+/**
+ * @type {webpack.Configuration}
+ */
+const webpackBaseConfig = {
```
Comment (clebert) [inline packages/demos/src/webpack-base-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/todomvc/webpack-config.js [hunk 3]
```diff
@@ -75,13 +86,19 @@ const configs: Configuration[] = [
...
+      new CopyPlugin([
+        path.join(__dirname, './index.css'),
+        path.join(__dirname, './index.html')
+      ])
+    ]
   }
 ];
 
-export default configs;
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/todomvc/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/todomvc/webpack-config.js [hunk 4]
```diff
@@ -1,46 +1,57 @@
...
-import {Configuration} from 'webpack';
-import merge from 'webpack-merge';
-import {webpackBaseConfig} from '../webpack-base-config';
+const postcssPresetEnv = require('postcss-preset-env');
+const webpack = require('webpack');
+const merge = require('webpack-merge');
+const webpackBaseConfig = require('../webpack-base-config');
 
-const configs: Configuration[] = [
+const websiteBuildDir = path.resolve(
```
Comment (clebert) [inline packages/demos/src/todomvc/webpack-config.js]: rename to `websiteDirname` or `websiteBuildDirname`

File: packages/demos/src/server-side-rendering/webpack-config.js [hunk 5]
```diff
@@ -36,4 +43,6 @@ export default [
       publicPath: '/'
     }
   }
-] as Configuration[];
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/server-side-rendering/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/module-loader-commonjs/webpack-config.js [hunk 6]
```diff
@@ -21,4 +25,6 @@ export default [
       publicPath: '/'
     }
   }
-] as Configuration[];
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/module-loader-commonjs/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/module-loader-amd/webpack-config.js [hunk 7]
```diff
@@ -23,4 +27,6 @@ export default [
       publicPath: '/'
     }
   }
-] as Configuration[];
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/module-loader-amd/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/integrator-dom/webpack-config.js [hunk 8]
```diff
@@ -20,4 +24,6 @@ export default [
       publicPath: '/'
     }
   }
-] as Configuration[];
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/integrator-dom/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/history-service/webpack-config.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    ...webpackBaseConfig,
+    entry: join(__dirname, './integrator.tsx'),
+    output: {
+      filename: 'integrator.js',
+      publicPath: '/'
+    }
+  }
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/history-service/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/feature-app-in-feature-app/webpack-config.js [hunk 10]
```diff
@@ -37,4 +44,6 @@ export default [
       publicPath: '/'
     }
   }
-] as Configuration[];
+];
+
+module.exports = configs;
```
Comment (clebert) [inline packages/demos/src/feature-app-in-feature-app/webpack-config.js]: we could also make a direct assignment to `module.exports`

File: packages/demos/src/webpack-base-config.js (hunk 2 above)
Comment (KingHenne) [reply packages/demos/src/webpack-base-config.js]: Last time I checked, this didn’t work with the type annotation, will check again tomorrow.

File: packages/demos/src/todomvc/webpack-config.js (hunk 1 above)
Comment (KingHenne) [reply packages/demos/src/todomvc/webpack-config.js]: Because I also use `resolve` here, and I like to have the `path` context when reading the call site. Therefore, I would suggest we never restructure from `path`. This is also already the case in some of the other files, e.g.`process-api-docs.js`.

File: packages/demos/src/webpack-base-config.js (hunk 2 above)
Comment (KingHenne) [reply packages/demos/src/webpack-base-config.js]: This is still the case, so unfortunately we need the `const`.

File: packages/demos/src/todomvc/webpack-config.js (hunk 4 above)
Comment (KingHenne) [reply packages/demos/src/todomvc/webpack-config.js]: done
